pom.xml

```diff
@@ -72,21 +72,21 @@
 
   <properties>
     <javaVersion>8</javaVersion>
-    <mavenVersion>3.2.5</mavenVersion>
+    <mavenVersion>3.9.6</mavenVersion>
```

Comment:
This lifts API to 3.9.6, isn't that a contradiction?

Reply:
Contradiction to what? The CI is updated by Slawek, so is build using 3.6.3/3.9.3 (see matrix), so min/max Maven is used, and deprecations immediately popped out (and were fixed as well).

Reply:
One is build, one API, no? Thought we raise at least build requirement to 3.6.3...

Reply:
If we'd raise (well, lower, as this PR raises to 3.9.6) build to 3.6.3 we'd see deprecations from 2019, those from 5 years ago, no? We should build against latest, to see all current deprecations, and backward compat allows us to state (and test it) even with older maven versions, like CI shows (3.6.3 used as well). Unsure what is the question actually...

Reply:
Let me think about that tomorrow or so...